code/game/machinery/doors/firedoor.dm

```diff
@@ -61,18 +63,13 @@
 	. = ..()
 	COOLDOWN_START(src, detect_cooldown, DETECT_COOLDOWN_STEP_TIME)
 	soundloop = new(src, FALSE)
-	AddElement(/datum/element/atmos_sensitive, mapload)
 	CalculateAffectingAreas()
 	my_area = get_area(src)
-	var/static/list/loc_connections = list(
-		COMSIG_TURF_EXPOSE = .proc/check_atmos,
-	)
-
-	AddElement(/datum/element/connect_loc, loc_connections)
 	if(!merger_typecache)
 		merger_typecache = typecacheof(/obj/machinery/door/firedoor)
-
-	check_atmos()
+	RegisterSignal(src, COMSIG_MERGER_ADDING, .proc/merger_adding)
+	RegisterSignal(src, COMSIG_MERGER_REMOVING, .proc/merger_removing)
+	GetMergeGroup(merger_id, merger_typecache)
```

Comment:
Do this in late initialize, that way things will spread out to their neighbors in one go, rather then creating and remaking merge groups over and over again